pkg/allocation/converters/converter.go

```diff
@@ -27,17 +27,21 @@ import (
 )
 
 // ConvertAllocationRequestToGSA converts AllocationRequest to GameServerAllocation V1 (GSA)
-func ConvertAllocationRequestToGSA(in *pb.AllocationRequest) *allocationv1.GameServerAllocation {
+func ConvertAllocationRequestToGSA(in *pb.AllocationRequest) (*allocationv1.GameServerAllocation, error) {
 	if in == nil {
-		return nil
+		return nil, nil
+	}
+
+	// nolint:staticcheck
+	if len(in.GameServerSelectors) != 0 && in.RequiredGameServerSelector != nil {
```

Comment:
Shouldn't there be a similar check (and test in `main_test.go`) for `len(in.GameServerSelectors) != 0 && len(in.PreferredGameServerSelector) != 0` to validate that the legacy preferred field is not set if the new selectors field is set?

Reply:
Done.